stockfish.py

```diff
@@ -0,0 +1,144 @@
+""" All UCI documentation here: http://wbec-ridderkerk.nl/html/UCIProtocol.html """
+
+import subprocess
+from typing import Dict, List, Optional, Union
+
+
+class Stockfish:
+    """ Integrates the Stockfish chess engine with Python. """
+
+    def __init__(
+        self, path: str = None, depth: int = 2, param: Dict[str, Union[str, int]] = None
+    ):
+        if param is None:
+            param = {}
+        if path is None:
+            path = "stockfish"
+        self.stockfish = subprocess.Popen(
+            path, universal_newlines=True, stdin=subprocess.PIPE, stdout=subprocess.PIPE
+        )
+        self.depth = str(depth)
+
+        # Tell the engine to use the UCI format
+        self.__put("uci")
+
+        default_param = {
+            "Write Debug Log": "false",
+            "Contempt": 0,
+            "Min Split Depth": 0,
+            "Threads": 1,
+            "Ponder": "false",
+            "Hash": 16,
+            "MultiPV": 1,
+            "Skill Level": 20,
```

Comment:
Is this a value we can modify and pass into the engine to change the difficulty?